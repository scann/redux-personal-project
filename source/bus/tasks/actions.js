//Types
import { types } from './types';

export const tasksActions = {
    //Sync
    fillTasks: (tasks) => {
        return {
            type:    types.FILL_TASKS,
            payload: tasks,
        };
    },
    createTask: (message) => {
        return {
            type:    types.CREATE_TASK,
            payload: message,
        };
    },

    //Async
    fetchTasksAsync: () => {
        return {
            type: types.FETCH_TASKS_ASYNC,
        };
    },
    createTaskAsync: (message) => {
        return {
            type:    types.CREATE_TASK_ASYNC,
            payload: message,
        };
    },
};
