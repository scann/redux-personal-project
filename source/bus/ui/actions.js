//Types
import { types } from './types';

export const uiActions = {
    startSpinning: () => {
        return {
            type: types.START_SPINNING,
        };
    },
    stopSpinning: () => {
        return {
            type: types.STOP_SPINNING,
        };
    },
    emitError: (error, meta = null) => {
        return {
            type:    types.EMIT_ERROR,
            payload: error,
            error:   true,
            meta,
        };
    },
    updateTasksFilter: (filter) => {
        return {
            type:    types.UPDATE_TASKS_FILTER,
            payload: filter,
        };
    },
    startEditingTask: (id, message) => {
        return {
            type:    types.START_EDITING_TASK,
            payload: {
                id,
                message,
            },
        };
    },
    resetEditingTask: () => {
        return {
            type: types.RESET_EDITING_TASK,
        };
    },
    confirmEditingTask: (taskId, message) => {
        return {
            type:    types.CONFIRM_EDITING_TASK,
            payload: {
                taskId,
                message,
            },
        };
    },
    updateTaskMessage: (message) => {
        return {
            type:    types.UPDATE_TASK_MESSAGE,
            payload: message,
        };
    },
};
