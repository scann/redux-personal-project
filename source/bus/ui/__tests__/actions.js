//Actions
import { uiActions } from '../actions';

describe('ui actions:', () => {
    test('startSpinning', () => {
        expect(uiActions.startSpinning()).toMatchSnapshot();
    });

    test('stopSpinning', () => {
        expect(uiActions.stopSpinning()).toMatchSnapshot();
    });

    test('emitError', () => {
        expect(uiActions.emitError(__.error)).toMatchSnapshot();
    });

    test('updateTasksFilter', () => {
        expect(uiActions.updateTasksFilter(__.filter)).toMatchSnapshot();
    });

    test('startEditingTask', () => {
        expect(uiActions.startEditingTask(__.testMessage.id, __.testMessage.message)).toMatchSnapshot();
    });

    test('resetEditingTask', () => {
        expect(uiActions.resetEditingTask()).toMatchSnapshot();
    });

    test('confirmEditingTask', () => {
        expect(uiActions.confirmEditingTask(__.testMessage.id, __.testMessage.message)).toMatchSnapshot();
    });

    test('updateTaskMessage', () => {
        expect(uiActions.updateTaskMessage(__.testMessage.message)).toMatchSnapshot();
    });
});
