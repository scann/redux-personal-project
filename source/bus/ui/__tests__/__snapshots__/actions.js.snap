// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: confirmEditingTask 1`] = `
Object {
  "payload": Object {
    "message": "TEST_MESSAGE",
    "taskId": "TEST_ID",
  },
  "type": "CONFIRM_EDITING_TASK",
}
`;

exports[`ui actions: emitError 1`] = `
Object {
  "error": true,
  "meta": null,
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions: resetEditingTask 1`] = `
Object {
  "type": "RESET_EDITING_TASK",
}
`;

exports[`ui actions: startEditingTask 1`] = `
Object {
  "payload": Object {
    "id": "TEST_ID",
    "message": "TEST_MESSAGE",
  },
  "type": "START_EDITING_TASK",
}
`;

exports[`ui actions: startSpinning 1`] = `
Object {
  "type": "START_SPINNING",
}
`;

exports[`ui actions: stopSpinning 1`] = `
Object {
  "type": "STOP_SPINNING",
}
`;

exports[`ui actions: updateTaskMessage 1`] = `
Object {
  "payload": "TEST_MESSAGE",
  "type": "UPDATE_TASK_MESSAGE",
}
`;

exports[`ui actions: updateTasksFilter 1`] = `
Object {
  "payload": "FILTER_STRING",
  "type": "UPDATE_TASKS_FILTER",
}
`;
